### ISSUE TITLE
Publishing: Images do not show up with simplest publishing technique

### ISSUE THREAD

**Issue by nickolay-kondratyev** [dendronhq/dendron#2698]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [ ] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [x] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

Images that are within the repository do NOT get shown upon publishing using the simplest method of publishing (running a local server). 

## To Reproduce

Create a workspace with a `vault` that contains `vault/assets/images/my-image.png` 

Create a note `note.md` that has the following content 

```
![img](assets/images/my-image.png)
```

Validate that the preview of `note.md` shows the image.

Go through the steps described in publish wiki https://wiki.dendron.so/notes/e5st4LFLtIwwbQmC6JBaF/

And after running the `npx dendron publish export` 

Grab the files from `.next/out/` and utilize python server to serve these files:

```
python -m SimpleHTTPServer 8000
```

## Expected behavior

Image is shown in the published note. 

Actual behavior: The site is being served by ```python -m SimpleHTTPServer 8000``` (look up, searching works). But the images are shown as broken links. 
 
## Screenshots

<img width="175" alt="Screen Shot 2022-04-05 at 6 33 11 PM" src="https://user-images.githubusercontent.com/4050134/161877872-600d151d-3b12-4a7c-bc34-705fcd978060.png">

## Desktop (please complete the following information)

- Browser (if with published websites; include version): [e.g. chrome, safari]
- Versions (all that apply to bug):
  - Dendron Extension: 88.0
  - VS Code version:
  - Dendron CLI:

**Comment by nickolay-kondratyev** [comment]
If some other simple server method works better than `python SimpleHTTPServer` (node etc..), would be happy to use it.

Also could be good to add this as one of the ways to serve the published content.

**Comment by jonathanyeung** [comment]
@nickolay-kondratyev - thanks for reporting.  Do you see the image assets getting copied over into `.next/out/` after running the `npx dendron publish export` command? @kevinslin - do you know if we have any existing issues around this that we're tracking?

**Comment by nickolay-kondratyev** [comment]
@jonathanyeung Yes the image assets are copied under `.next/out/`

`.next/out/assets/images` is full of images.

**Comment by jonathanyeung** [comment]
Hrm - unable to repro thus far. 

Are you seeing 404's in the python simple server output? 

I just tried a basic example with python simpleHTTPServer and so far it's rendering the image ok. 

```log
...
127.0.0.1 - - [07/Apr/2022 17:16:23] "GET /data/dendron.json HTTP/1.1" 200 -
127.0.0.1 - - [07/Apr/2022 17:16:23] "GET /_next/static/chunks/213.0b43b5c507169128.js HTTP/1.1" 200 -
127.0.0.1 - - [07/Apr/2022 17:16:23] "GET /_next/static/chunks/798.fae27e19f205f4ee.js HTTP/1.1" 200 -
127.0.0.1 - - [07/Apr/2022 17:16:25] "GET /_next/static/chunks/pages/notes/%5Bid%5D-dc9b1cbe134a7143.js HTTP/1.1" 200 -
127.0.0.1 - - [07/Apr/2022 17:16:25] "GET /_next/data/Lqyp6D65xHBWEStQqWMPC/notes/arqdtyeksn7gmm90hhq1l1d.json HTTP/1.1" 200 -
127.0.0.1 - - [07/Apr/2022 17:16:25] "GET /assets/images/2022-04-06-18-02-46.png HTTP/1.1" 200 -
127.0.0.1 - - [07/Apr/2022 17:16:26] "GET /_next/data/Lqyp6D65xHBWEStQqWMPC/index.json HTTP/1.1" 200 -
127.0.0.1 - - [07/Apr/2022 17:16:31] "GET /_next/data/Lqyp6D65xHBWEStQqWMPC/notes/87ryid2vvx48yxi5m8x16co.json HTTP/1.1" 200 -
```

**Comment by nickolay-kondratyev** [comment]
Yes seeing 404s in this example `gtlz9qnzpobtwfvj5ha5g88` is the id of the note being renderred.
```
10.187.170.15 - - [12/Apr/2022 02:55:26] code 404, message File not found
10.187.170.15 - - [12/Apr/2022 02:55:26] "GET /notes/gtlz9qnzpobtwfvj5ha5g88/assets/images/lambda_dlq_ddb_stream_src.png HTTP/1.1" 404 -
```

I tried to setup brand new vault and in that vault I am actually able to see the images.

**Comment by nickolay-kondratyev** [comment]
@jonathanyeung are you trying the basic example from `root.md` note? 

It appears that for me if I add an image to `root.md` the image is rendered but if I add the image to a note right next to root for example `hi.md` then the same image does not render anymore. 

Due to 404 such as 
```
10.187.170.15 - - [12/Apr/2022 03:17:34] code 404, message File not found
10.187.170.15 - - [12/Apr/2022 03:17:34] "GET /notes/9y0v9nlt6xs55ek4pb3klfe/assets/images/dendron_root_image.png.png HTTP/1.1" 404 -
```

Where `9y0v9nlt6xs55ek4pb3klfe` is the id of the `hi.md` note.

**Comment by nickolay-kondratyev** [comment]
Also appears to reproducible if adding contents of `out` file to S3 bucket, `root` is able to render the image. But `hi` note does not render the image.

**Comment by nickolay-kondratyev** [comment]
@jonathanyeung Just wanted to make sure you are able to reproduce this now right?

**Comment by jonathanyeung** [comment]
@nickolay-kondratyev - Haven't been able to repro this exactly, but I have been noticing asset rendering issues happening intermittently in various scenarios.  Still trying to chase down the root cause. 

Can you share with me your publishing config, i.e. the `publishing` section of `dendron.yml`?

**Comment by nickolay-kondratyev** [comment]
Here is the entire test repository with, log notes, environment recorded, and entire `.next/out` folder as well to record as much state as possible. Let me know if you are able to reproduce it from running from this

https://drive.google.com/file/d/1qUyh66qSsvlQ5nD4PbSRXCYON3kIYy5g/view?usp=sharing


Run this command to unpack the folder 

```
untarUngzFile ()
{
    tar xzf "$1"
}
```

And place it into `/tmp/ForPublishTest/` for the other command specified in [[command-used-for-publishing]] to work.

**Comment by jonathanyeung** [comment]
Thanks @nickolay-kondratyev - I can repro with your example.  Let me continue investigating.

**Comment by jonathanyeung** [comment]
@nickolay-kondratyev - I think there's a simple workaround: prefix all your image links with a slash: 
```
![](/assets/images/screenshot.png)
```

instead of 

```
![](assets/images/screenshot.png)
```

We have some logic in preview to make it work even if there's no leading slash, but it looks like that logic got dropped in the publishing flow.  We'll follow up on that part.

**Comment by nickolay-kondratyev** [comment]
@jonathanyeung Thanks! Yea it looks like adding a `/` at beginning fixes the issue.

**Comment by jonathanyeung** [comment]
Related Fix that should permit assets without a leading `/`: https://github.com/dendronhq/dendron/pull/2899

**Comment by Harshita-mindfire** [comment]
This should be fixed after the Dendron v0.95 Feel free to comment here if that's not the case.
